File: orchalang-spring-boot-sample-app/src/main/orcha/configuration/transport/ProductOrderConfiguration.java
package configuration.transport;

import orcha.lang.configuration.*;
import org.springframework.context.annotation.Bean;
import service.transport.ProfileUser;

public class ProductOrderConfiguration {

    @Bean
    public EventHandler user(){
        EventHandler eventHandler = new EventHandler("user");
        InputFileAdapter inputFileAdapter = new InputFileAdapter("data/input", "request.json");
        eventHandler.setInput(new Input("application/json", "service.transport.Request", inputFileAdapter));
        OutputFileAdapter outputFileAdapter = new OutputFileAdapter("data/output", true, "securityAlert.json", true, OutputFileAdapter.WritingMode.REPLACE);
        eventHandler.setOutput(new Output("application/json", "service.airport.SecurityAlert", outputFileAdapter));
        return eventHandler;
    }

    @Bean
    public ProfileUser profileUser() {
        return new ProfileUser();
    }

    @Bean
    public Application authenticateUsersRequest(){
        Application application = new Application("authenticateUsersRequest", "Java");
        JavaServiceAdapter javaAdapter = new JavaServiceAdapter("service.transport.ProfileUser", "check");
        application.setInput(new Input("service.transport.Request", javaAdapter));
        application.setOutput(new Output("service.transport.Request", javaAdapter));
        return application;
    }

}

File: orchalang-spring-boot-sample-app/src/main/orcha/configuration/transport/ProductOrderConfiguration.kt
package configuration.transport

import orcha.lang.configuration.*
import org.springframework.context.annotation.Bean
import service.transport.ProfileUser

class ProductOrderConfiguration {

    @Bean
    fun user(): EventHandler {
        val eventHandler = EventHandler("user")
        val inputFileAdapter = InputFileAdapter("data/input", "request.json")
        eventHandler.input = Input("application/json", "service.transport.Request", inputFileAdapter)
        val outputFileAdapter = OutputFileAdapter("data/output", true, "securityAlert.json", true, OutputFileAdapter.WritingMode.REPLACE)
        eventHandler.output = Output("application/json", "service.airport.SecurityAlert", outputFileAdapter)
        return eventHandler
    }

    @Bean
    fun profileUser(): ProfileUser {
        return ProfileUser()
    }

    @Bean
    fun authenticateUsersRequest(): Application {
        val application = Application("authenticateUsersRequest", "Java")
        val javaAdapter = JavaServiceAdapter("service.transport.ProfileUser", "check")
        application.input = Input("service.transport.Request", javaAdapter)
        application.output = Output("service.transport.Request", javaAdapter)
        return application
    }

}
